packages/next/src/server/app-render/action-handler.ts

```diff
@@ -209,26 +209,16 @@ async function createRedirectRenderResult(
     // }
 
     try {
-      const headResponse = await fetch(fetchUrl, {
-        method: 'HEAD',
+      const response = await fetch(fetchUrl, {
+        method: 'GET',
         headers: forwardedHeaders,
         next: {
           // @ts-ignore
           internal: 1,
         },
       })
 
-      if (
-        headResponse.headers.get('content-type') === RSC_CONTENT_TYPE_HEADER
-      ) {
-        const response = await fetch(fetchUrl, {
-          method: 'GET',
-          headers: forwardedHeaders,
-          next: {
-            // @ts-ignore
-            internal: 1,
-          },
-        })
+      if (response.headers.get('content-type') === RSC_CONTENT_TYPE_HEADER) {
```

Comment:
If the header doesn't match, should we should either consume the response body (there's a [memory leak](https://twitter.com/cramforce/status/1762142087930433999)) or abort the request for performance?

Reply:
Good catch, yes!